### PR TITLE
Throw an exception if a model relation is incomplete

### PR DESCRIPTION
**Pull request by aschempp**
Fixes https://github.com/contao/contao/issues/5818

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]

**Review by fritzmg — COMMENTED** [review]
How can this be reproduced? If you did not define a table in your `relation` or `foreignKey` config on your DCA field, then the `Incomplete relation defined for tl_foobar.foobar` exception is thrown.

**Comment by aschempp** [comment]
I don't know exactly, but reading the code tells me there's a missing check for a non-existing value here. Regardless on how the data is built 🤷

**Comment by fritzmg** [comment]
In any case the original issue seems to have a different cause entirely.

File: core-bundle/src/Resources/contao/library/Contao/Model.php [hunk 1]
```diff
@@ -152,6 +152,12 @@ public function __construct($objResult=null)
 			// Create the related models
 			foreach ($arrRelated as $key=>$row)
 			{
+				if (!isset($this->arrRelations[$key]['table']))
+				{
+					$this->arrRelated[$key] = null;
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Model.php]: Shouldn‘t we throw an exception here instead? Because this case should never occur.

Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/Model.php]: I guess both works 🤷 but we're setting the relation to `null` a few lines below as well

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Model.php]: Yes, but that is an expected case. This one is not, so we should not discard it silently.

Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/Model.php]: changed in b40ab81779b7b9b5524a4eb3059c48ab80d69ea9

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**